azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/AppInsightsProxy.java

```diff
@@ -75,7 +82,7 @@ private String readInstrumentationKeyFromConfiguration() {
             } else {
                 return StringUtils.EMPTY;
             }
-        } catch (IOException exception) {
+        } catch (IOException | NullPointerException exception) {
```

Comment:
Don't catch NullPointerException to detect absent of file, use `if (inputStream == null) ` instead.

Reply:
Fix it.